lmcache/v1/lookup_client/lmcache_lookup_client.py

```diff
@@ -38,13 +39,26 @@
 class LMCacheLookupClient(LookupClientInterface):
     """ZMQ-based lookup client that communicates with a lookup server."""
 
-    def __init__(self, vllm_config: "VllmConfig"):
+    def __init__(self, vllm_config: "VllmConfig", config: LMCacheEngineConfig):
         self.encoder = MsgpackEncoder()
         self.ctx = zmq.Context()  # type: ignore[attr-defined]
         rpc_port = vllm_config.kv_transfer_config.get_from_extra_config(
             "lmcache_rpc_port", 0
         )
         self.tensor_parallel_size = vllm_config.parallel_config.tensor_parallel_size
+        self.create_lookup_server_only_on_worker_0 = (
+            config.extra_config
+            and config.extra_config.get("create_lookup_server_only_on_worker_0", True)
+        )
+        if self.create_lookup_server_only_on_worker_0:
+            socket_path = get_zmq_rpc_path_lmcache(vllm_config, rpc_port)
+            self.socket = make_zmq_socket(
+                self.ctx,
+                socket_path,
+                zmq.REQ,  # type: ignore[attr-defined]
+                bind=False,
+            )
+            return
         for tp_rank in range(self.tensor_parallel_size):
```

Comment:
We can simplify the code here to avoid copy-pasting:

```suggestion
        ranks = self.tensor_parallel_size
        if self.create_lookup_server_only_on_worker_0:
            ranks = 1
        for tp_rank in range(ranks):
```